src/blk/kernel/KernelDevice.h

```diff
@@ -124,7 +124,7 @@ class KernelDevice : public BlockDevice,
 
   void aio_submit(IOContext *ioc) override;
   void discard_drain() override;
-
+  const interval_set<uint64_t>* get_discard_queued() override { return &discard_queued;}
```

Comment:
Relying on this doesn't take into account `discard_finishing` in `_discard_thread()`, which can be arbitrarily long